src/Swarm/App.hs

```diff
@@ -103,12 +111,18 @@ appMain opts = do
             handleEvent e
 
       -- Setup virtual terminal
-      let buildVty = V.mkVty $ V.defaultConfig {V.colorMode = colorMode opts}
-      initialVty <- buildVty
-      V.setMode (V.outputIface initialVty) V.Mouse True
+      let mkVty =
+            case colorMode opts of
+              Nothing    -> VS.mkVty V.defaultConfig
+              Just cMode -> do
+                    platformSettings <- VS.defaultSettings
+                    VS.mkVtyWithSettings V.defaultConfig $ platformSettings { VS.settingColorMode = cMode }
+      vty <- mkVty
```

Comment:
This is odd, why do we need to run `mkVty` twice? (Here and below in `customMain`.)

I noticed this because it would be nicer to have `vty <-` instead of `let mkVty =`, but now I wonder if running it twice could cause any issues with the terminal. 🤔 

If it has to be done this way, please add a comment with explanation.

Reply:
So it looks like it is called twice, but actually the function to construct a Vty instance is only called once. the 'mkVty' variable has type 'IO Vty' and is necessary because you are using the customMain function which requires a Vty instance, and an IO Vty (I don't know why...). 

Anyway, this is exactly what you were doing in this function before. I just changed mkVty to mkVtyWithSettings when the colorMode is available. Otherwise, use the old mkVty.